src/nvim/memline.c

```diff
@@ -1360,9 +1360,10 @@ recover_names (
     /* check for out-of-memory */
     for (i = 0; i < num_names; ++i) {
       if (names[i] == NULL) {
-        for (i = 0; i < num_names; ++i)
-          free(names[i]);
+        for (j = 0; j < num_names; ++j)
+          free(names[j]);
         num_names = 0;
+        break;
```

Comment:
I believe the original code does its job, but I needed some time to figure out why it had been written this way. So yes, to make the intent clear.